src/sql/workbench/services/notebook/browser/outputs/tableRenderers.ts

```diff
@@ -88,13 +87,17 @@ export function renderDataResource(
 }
 
 // SlickGrid requires columns and data to be in a very specific format; this code was adapted from tableInsight.component.ts
-export function transformData(rows: any[], columns: Slick.Column<any>[]): { [key: string]: string }[] {
+function transformData(rows: IDataResourceRow[], columns: Slick.Column<any>[]): IDataResourceRow[] {
+	// Rows are either indexed by column name or ordinal number, so check for one column name to see if it uses that format
+	let useColumnNameKey = Object.keys(rows[0]).includes(columns[0].name);
```

Comment:
Consider making a helper function in case we want to update this logic later. 

Is there any concern about there being a column with a numeric name (e.g. `0`, `1`) and that messing up this check?

Reply:
Probably. I'll do a Set comparison between them to eliminate any ordering issues.